geo/Geoserver.py

```diff
@@ -48,20 +51,37 @@ class Geoserver:
 
     def __init__(
         self,
-        service_url="http://localhost:8080/geoserver",
-        username="admin",
-        password="geoserver",
+        service_url: str = "http://localhost:8080/geoserver/rest",  # default deployment url during installation
+        username: str = "admin",  # default username during geoserver installation
+        password: str = "geoserver",  # default password during geoserver installation
     ):
         self.service_url = service_url
         self.username = username
         self.password = password
 
+        # private request method to reduce repetition of putting auth(username,password) in all requests call. DRY principle
+
+    def _requests(self, method: str, url: str, **kwargs) -> requests.Response:
```

Comment:
This is very useful. We can integrate this method in the next PR/release.

Reply:
Alright.  Will do